albogdan/skule-vote
# Initial AWS CodeDeploy Setup

## Overview

- Resolves #90 
- Creates a bunch of files and folders to integrate with AWS CodeDeploy. From the top of the file structure:
    - Renames the `main.yml` workflow to `checks.yml` and creates a new `deploy.yml` workflow that is used to deploy to `AWS`.
    - Creates the `deployment/` folder which contains a `Dockerfile`, `docker-compose.prod.yml`, and `entrypoint.sh` which serve to host the `Django` backend server.
    - Creates the `scripts/` folder which contains:
        - `application_stop.sh` - stops the previous docker container that was running, if it exists (doesn't exist on the initial run)
        - `after_install.sh` - installs python requirements, builds and collects both `react` and `django` static files, and moves the static files to `/var/www/html`.
        - `start_server.sh` - builds and starts the new django backend server in a docker container.
        - `nginx/default` - the file that describes the default config for the nginx server that works with our setup. This file is how it looks before `CertBot` is run on it.
    - Creates the `appspec.yml` file which describes the order in which the scripts run. Note that before the `AfterInstall` stage, there is an `Install` stage which runs and copies the repo to `/home/ec2-user/skule-vote` - as seen in the `source` and `destination` parameters in the file.
- Created two new wiki pages that explain the setup for the [infrastructure](https://github.com/albogdan/skule-vote/wiki/First-Time-Infrastructure-Setup) and the [software](https://github.com/albogdan/skule-vote/wiki/First-Time-EC2-Software-Setup). Read the links for more information please.

**NOTE: this is only for staging and deploys anything that is pushed to develop. Will set up the one for main/production after we get everything up and running with this one.**

## Unit Tests Created

- None


## Steps to QA

- Read thru the [infrastructure](https://github.com/albogdan/skule-vote/wiki/First-Time-Infrastructure-Setup) and the [software](https://github.com/albogdan/skule-vote/wiki/First-Time-EC2-Software-Setup) docs mentioned above.
- Idk look thru the code it and see if it makes sense I guess? Ask me about anything that doesn't make sense.
